hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestOzoneFileInterfaces.java

```diff
@@ -217,27 +219,70 @@ private void verifyOwnerGroup(FileStatus fileStatus) {
 
   @Test
   public void testDirectory() throws IOException {
-    String dirPath = RandomStringUtils.randomAlphanumeric(5);
-    Path path = createPath("/" + dirPath);
-    assertTrue("Makedirs returned with false for the path " + path,
-        fs.mkdirs(path));
-
-    FileStatus status = fs.getFileStatus(path);
-    assertTrue("The created path is not directory.", status.isDirectory());
-
-    assertTrue(status.isDirectory());
-    assertEquals(FsPermission.getDirDefault(), status.getPermission());
-    verifyOwnerGroup(status);
+    String leafName = RandomStringUtils.randomAlphanumeric(5);
+    OMMetadataManager metadataManager = cluster.getOzoneManager()
+        .getMetadataManager();
+
+    String lev1dir = "abc";
+    Path lev1path = createPath("/" + lev1dir);
+    String lev1key = metadataManager.getOzoneDirKey(volumeName, bucketName,
+        o3fs.pathToKey(lev1path));
+    String lev2dir = "def";
+    Path lev2path = createPath("/" + lev1dir + "/" + lev2dir);
+    String lev2key = metadataManager.getOzoneDirKey(volumeName, bucketName,
+        o3fs.pathToKey(lev2path));
+
+    FileStatus rootChild;
+    FileStatus rootstatus;
+    FileStatus leafstatus;
+
+    Path leaf = createPath("/" + lev1dir + "/" + lev2dir + "/" + leafName);
+    String leafKey = metadataManager.getOzoneDirKey(volumeName, bucketName,
+        o3fs.pathToKey(leaf));
+
+    // verify prefix directories and the leaf, do not already exist
+    assertTrue(metadataManager.getKeyTable().get(lev1key) == null);
+    assertTrue(metadataManager.getKeyTable().get(lev2key) == null);
+    assertTrue(metadataManager.getKeyTable().get(leafKey) == null);
+
+    assertTrue("Makedirs returned with false for the path " + leaf,
+        fs.mkdirs(leaf));
+
+    // verify the leaf directory got created.
+    leafstatus = getDirectoryStat(leaf);
+    assertTrue(leafstatus != null);
+
+    if (cluster.getOzoneManager().createPrefixEntries()) {
+      FileStatus lev1status;
+      FileStatus lev2status;
+
+      // verify prefix directories got created when creating the leaf directory.
+      String kn = metadataManager.getKeyTable().get(lev2key).getKeyName();
+      assertTrue(kn.equals("abc/def/"));
+      assertTrue(metadataManager
+          .getKeyTable()
+          .get(lev1key)
+          .getKeyName().equals("abc/"));
+      assertTrue(metadataManager
+          .getKeyTable()
+          .get(lev2key)
+          .getKeyName().equals("abc/def/"));
+      lev1status = getDirectoryStat(lev1path);
+      lev2status = getDirectoryStat(lev2path);
+      assertTrue((lev1status != null) && (lev2status != null));
+      rootChild = lev1status;
+    } else {
```

Comment:
This case would never be reached in this test suit as createPrefix config is true by default and we are not overriding it.

Reply:
I'll fix that in a subsequent patch. this is a comment from @mukul1987 as well. we will include both cases in the parametrised test.